docs: Update Android Kotlin SDK documentation for improved Compose auto-capture (#792)
amplitude/amplitude-docs: content/collections/android_sdk/en/android-kotlin-sdk.md

{{/partial:collapse}}

{{partial:admonition type="info" heading="Support for Jetpack Compose"}}
Amplitude automatically tracks user interactions with all clickable UI elements implemented in Jetpack Compose. The `Modifier.testTag` is optional and can be added to `@Composable` functions to provide additional identification in the `[Amplitude] Target Tag` property. If no `testTag` is provided, the element will still be tracked with other available properties.

**Using testTag for Better Element Identification**

While `testTag` is optional, it's highly recommended for identifying specific Compose views that were clicked. The `testTag` property provides several benefits:

> - **Precise Element Identification**: Helps you distinguish between similar UI elements (like multiple buttons or cards) in your analytics data

Reviewer: ```suggestion
- **Precise Element Identification**: Helps distinguish between similar UI elements (like multiple buttons or cards) in your analytics data.
```

- **Stable Tracking**: Provides a consistent identifier that won't change when UI structure or styling is modified

> - **Easier Analysis**: Makes it simpler to filter and analyze interactions with specific elements in Amplitude charts

Reviewer: ```suggestion
- **Easier Analysis**: Enables easier filtering and analysis of interactions with specific elements in Amplitude charts.
```

> - **Cross-Platform Consistency**: Allows you to maintain consistent element naming across different platforms

Reviewer: ```suggestion
- **Cross-Platform Consistency**: Helps you to maintain consistent element naming across different platforms.
```

```kotlin
// Example: Adding testTag for better identification
Button(
    onClick = { /* handle click */ },
    modifier = Modifier.testTag("login_button")
) {
    Text("Log In")
}

Card(
    onClick = { /* handle click */ },
    modifier = Modifier.testTag("product_card_${product.id}")
) {
    // Card content
}
```

> When these elements are clicked, the `[Amplitude] Target Tag` property will contain the `testTag` value, making it easy to identify which specific element was interacted with in your analytics data.

Reviewer: ```suggestion
When a user clicks these elements, the `[Amplitude] Target Tag` property contains the `testTag` value, making it easy to identify which specific element was interacted with in your analytics data.
```
